### PR TITLE
Reposition cursor when pasting full card numbers

### PR DESCRIPTION
**Pull request by ddayguerrero**
### Changes

Resolves a couple of minor of issues when copy pasting full card numbers:

- When pasting a full length credit card, the cursor is not positioned at the very end, but instead, a few digits before. (e.g. pasting a (16) digit credit card will cause the cursor to be 3 digits before)

- When selecting a full length credit card, by mouse or keyboard, and then pasting again the same credit card number, it will add three extra numbers to the very end of it, resulting in a 19 digit card.

### Why introduce these changes?

- These cursor and pasting issues are re we plan to update our BIN regex patterns for our supported credit card numbers

### How is this achieved?

...

### CODE REVIEW

File: src/payform.coffee [hunk 1]
```diff
@@ -209,11 +209,20 @@
 
   reFormatCardNumber = (e) ->
     return if e.target.value is ""
+    if document.dir == 'ltr'
```
Comment (glebm) [inline src/payform.coffee]: `dir` may be also set via CSS, and/or on a particular element as it doesn't have to be the same for the whole document. Consider this instead:

```coffeescript
dir = getComputedStyle(e.target)['direction']
```

Comment (jondavidjohn) [reply src/payform.coffee]: Thoughts on this @ddayguerrero ?

Comment (ddayguerrero) [reply src/payform.coffee]: I had raised this concern when I first started introducing these conditionals for document directionality. 

I will take a deeper look on this, but it seems like a viable solution and also adds more flexibility!

https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle (Support for IE9+!!!)

**Comment by ddayguerrero** [comment]
@glebm your suggestion made sense and allows users to provide more specific functionality when using Payfrom in a RTL context. Do you mind giving this a try?

I created a fallback to the global directionality of the page in case the computer style object is `undefined`.

File: src/payform.coffee [hunk 2]
```diff
@@ -167,6 +167,10 @@
   cardFromType = (type) ->
     return card for card in payform.cards when card.type is type
 
+  getDirectionality = (target) ->
+    style = getComputedStyle(target)
+    return if style then style['direction'] else document.dir
```
Comment (glebm) [inline src/payform.coffee]: The fallback is only necessary on Firefox for some edge cases (such as a `display: none` iframe), https://bugzilla.mozilla.org/show_bug.cgi?id=1467722.

However, starting from Firefox 62, in the aforementioned edge cases Firefox no longer returns `null`: https://www.fxsitecompat.com/en-CA/docs/2018/getcomputedstyle-no-longer-returns-null-when-style-can-t-be-retrieved/

Instead, it returns an object with `length: 0` and an empty string for every declaration.

For Firefox 62+ compatibility, this should be something like:

```coffeescript
# Work around Firefox not returning the styles in some edge cases.
style and style['direction'] or document.dir
```

**Comment by glebm** [comment]
> Do you mind giving this a try?

Sorry but I'm not currently using Payform in an RTL context.

**Pull request merged**